src/DataStructures/DataBox/DataBoxTag.hpp

```diff
@@ -558,7 +560,43 @@ template <class Tag>
 struct remove_all_prefixes_impl<Tag, true> {
   using type = remove_all_prefixes<remove_tag_prefix<Tag>>;
 };
-}  // namespace databox_detail
+
+// Implementation of variables_tag_with_tags_list
+template <DispatchTagType TagType>
+struct variables_tag_with_tags_list_impl;
+}  // namespace DataBox_detail
+
+/// \ingroup DataBoxGroup
+/// Change the tags contained in a possibly prefixed Variables tag.
+template <typename Tag, typename NewTagsList>
+using variables_tag_with_tags_list =
```

Comment:
How about the shorter name `set_tags_list`? At the call site it reads `db::set_tags_list<VariablesTag, TagsList>` then, which is nicely readable.

Reply:
I strongly oppose that. This is in the databox namespace, so if it's not operating on a DataBox it should say so in the name.

Reply:
`set_variables_tags_list`?

Reply:
It doesn't "set" it, it generates a new tag with the new tag list. I'm currently unable to think of anything better than what Will has that's also shorter...

Reply:
that's fine then :)